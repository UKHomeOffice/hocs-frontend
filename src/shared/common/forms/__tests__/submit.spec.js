import React from 'react';
import Submit from '../submit.jsx';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

describe('Form button component', () => {
    it('should render with default props', () => {
        expect(
            render(<Submit />)
        ).toMatchSnapshot();
    });
    it('should render disabled when isDisabled is passed', () => {
        expect(
            render(<Submit disabled={true} />)
        ).toMatchSnapshot();
    });
    it('should render with correct when label is passed', () => {
        expect(
            render(<Submit label="Submit my form" />)
        ).toMatchSnapshot();
    });
});
