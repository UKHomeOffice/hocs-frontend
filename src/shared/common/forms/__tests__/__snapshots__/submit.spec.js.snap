// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form button component should render disabled when isDisabled is passed 1`] = `
<input
  class="govuk-button"
  type="submit"
  value="Submit"
/>
`;

exports[`Form button component should render with additional styles when className is passed 1`] = `
<input
  class="govuk-button start"
  type="submit"
  value="Submit"
/>
`;

exports[`Form button component should render with correct when label is passed 1`] = `
<input
  class="govuk-button"
  type="submit"
  value="Submit my form"
/>
`;

exports[`Form button component should render with default props 1`] = `
<input
  class="govuk-button"
  type="submit"
  value="Submit"
/>
`;
